bin/node/src/chain_spec.rs

```diff
@@ -109,6 +109,9 @@ pub struct ChainParams {
     #[structopt(long)]
     pub millisecs_per_block: Option<u64>,
 
+    #[structopt(long)]
+    pub base_unit_creation_delay: Option<u64>,
```

Comment:
Effectively, we only use it after convertig it to `f64`, is there anything preventing us from changing its type globally to `f64`?

Reply:
You are right, however, when using `f64` the compiler screams:
```
the trait `WrapperTypeDecode` is not implemented for `f64`
```
for the piece of code below (from `primitives/src/lib.rs`):
```rust
sp_api::decl_runtime_apis! {
    pub trait AlephSessionApi
    {
        fn next_session_authorities() -> Result<Vec<AuthorityId>, ApiError>;
        fn authorities() -> Vec<AuthorityId>;
        fn session_period() -> u32;
        fn millisecs_per_block() -> u64;
        fn base_unit_creation_delay() -> f64;
    }
}
```

Reply:
Ugh, this compiler error is so confusing. It means that actually `Decode` is not implemented for `f64`, and it checks if `WrapperTypeDecode` is implemented for `f64` because we have

```
impl<T, X> Decode for X where
    T: Decode + Into<X>,
    X: WrapperTypeDecode<Wrapped = T>
```

Apparently, acccording to authors of `codec`, we don't need floating point types. And maybe they are right? why not store the delay as an unsigned integer denoting milliseconds?